sdks/java/core/src/main/java/org/apache/beam/sdk/values/PCollectionView.java

```diff
@@ -50,12 +51,20 @@
 
   /**
    * For internal use only.
+   *
+   * @deprecated use getViewFn()
    */
+  @Deprecated
   public T fromIterableInternal(Iterable<WindowedValue<?>> contents);
 
   /**
    * For internal use only.
    */
+  public ViewFn<Iterable<WindowedValue<?>>, T> getViewFn();
```

Comment:
At least for now, it is basically a dumb replacement for `fromIterableInternal` just to make the architecture and vocabulary match up. Eventually, I agree: only `ParDo` needs to be able to read this, as all the info in a `PCollectionView` is just a way of plumbing the spec via POJO. We might want to make it an abstract class so that it can have package-private accessors. Alternatively, we could dream about new APIs for side inputs so that we never create the intermediate `PValue`. We'll need to weigh the cost to users porting their pipelines.